test/gracefulstop_test.go

```diff
@@ -148,10 +149,16 @@ func TestGracefulStop(t *testing.T) {
 		allowCloseCh: make(chan struct{}),
```

Comment:
Are the "steps of this test" still valid after these changes?

Does the intent of this test change at all?  (Should this be a new test case instead of a modification to this one?)

Reply:
Yes the steps remain the same. The idea is to eliminate a race; previously we'd allow the client to read goaway after it sends something on the wire(not necessarily headers). Now, we allow it to read goaway only after the server has received the headers.